### PR TITLE
refactor(core): clean up circular dependencies

### PR DESCRIPTION
**Pull request by mhevery**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:342779806:BASE:342779845:1605587781883:393efd89)

File: packages/core/src/di/injector_compatibility.ts [hunk 1]
```diff
@@ -22,6 +21,7 @@ import {ValueProvider} from './interface/provider';
 import {Inject, Optional, Self, SkipSelf} from './metadata';
 
 
+
```
Comment (petebacondarwin) [inline packages/core/src/di/injector_compatibility.ts:24]: ?

**Review by petebacondarwin — APPROVED** [review]
LGTM.
I guess the lesson here is that while it is nice to group stuff together, we must be careful when what appear to be separate groups actually have complex mutual dependencies.

File: packages/core/src/di/injector_compatibility.ts (hunk 1 above)
Comment (mhevery) [reply packages/core/src/di/injector_compatibility.ts:24]: reverted

**Review by AndrewKushnir — APPROVED** [review]
Thanks for doing this cleanup Misko! 👍

File: packages/core/src/render/api_flags.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+/**
+ * Flags for renderer-specific style modifiers.
+ * @publicApi
+ */
+export enum RendererStyleFlags2 {
+  // TODO(misko): This needs to be refactored into a separate file so that it can be imported from
+  // `node_manipulation.ts` Currently doing the import cause resolution order to change and fails
+  // the tests. The work around is to have hard coded value in `node_manipulation.ts` for now.
```
[the remark this answers is not in the document]
Comment (mhevery) [reply packages/core/src/render/api_flags.ts:53]: The comment is independent, but turns out that with this refactoring I can fix the underlying issue and remove the comment. So I have done so.

[the remark this answers is not in the document]
Comment (AndrewKushnir) [reply packages/core/src/render/api_flags.ts:53]: Thanks Misko 👍

**Review by IgorMinar — APPROVED** [review]
yay! more deps gone. thanks

Reviewed-for: global-approvers

**Pull request closed without merging**